### PR TITLE
Updates to Flux v10

### PR DESCRIPTION
**Pull request by DhairyaLGandhi**
Problems:

vision:
  - [ ] cifar OOMs (cc @maleadt so he is in the loop)

text:
  - [ ] phonemes - Mutation error (cc @MikeInnes any obvious place that jumps out?)
  - [ ] char-rnn - on GPU - `ERROR: LoadError: MethodError: no method matching Float32(::ForwardDiff.Dual{Nothing,Bool,1})` 

tutorials:
  - [ ] Scalar indexing - likely in accuracy function

Haven't checked `contrib` very closely, would be awesome to get some help on there

cc @MikeInnes @CarloLucibello

### CODE REVIEW

**Comment by dhairyagandhi96** [comment]
Other than that things seem fine. It is frustrating to get errors, which this time had a lot to do with `ImageIO` being needed explicitly in the environment.

File: tutorials/60-minute-blitz.jl [hunk 1]
```diff
@@ -273,9 +273,9 @@ Flux.train!(loss, params(m), [(data,labels)], opt)
 # It also has a number of dataloaders that come in handy to load datasets.
 
 using Statistics
-#using CuArrays
-using Zygote
+using Flux.Zygote
```
Comment (MikeInnes) [inline tutorials/60-minute-blitz.jl:276]: Can we just add Zygote to the project file?

File: tutorials/60-minute-blitz.jl [hunk 2]
```diff
@@ -273,9 +273,9 @@ Flux.train!(loss, params(m), [(data,labels)], opt)
...
 using Flux, Flux.Optimise
+# using Flux.CuArrays
```
Comment (MikeInnes) [inline tutorials/60-minute-blitz.jl:278]: I guess we can just remove these lines entirely, since CuArrays is always loaded anyway.

**Comment by CarloLucibello** [comment]
closing as outdated

**Pull request closed without merging**